### PR TITLE
Move property path code to separate file

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This is a pure refactor and just moves the propertyPath-related code to a new file. This is a separate PR to make reviewing easier.

Stacked on https://github.com/pulumi/pulumi-terraform-bridge/pull/2515, https://github.com/pulumi/pulumi-terraform-bridge/pull/2516 and https://github.com/pulumi/pulumi-terraform-bridge/pull/2496

### CODE REVIEW

**Comment by VenelinMartinov** [comment]
This change is part of the following stack:

- #2497 ◀

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

**Review by iwahbe — APPROVED** [review]

File: pkg/tfbridge/property_path.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+func lookupSchemas(
+	path propertyPath, tfs shim.SchemaMap, ps map[string]*info.Schema,
+) (shim.Schema, *info.Schema, error) {
+	schemaPath := PropertyPathToSchemaPath(resource.PropertyPath(path), tfs, ps)
+	return LookupSchemas(schemaPath, tfs, ps)
+}
+
+// walkPropertyValue walks a property value and calls the visitor function for each path in the property value.
+func walkPropertyValue(
```
Comment (t0yv0) [inline pkg/tfbridge/property_path.go:54]: We have some walkers for PropertyValue elsewhere.

https://github.com/pulumi/pulumi-terraform-bridge/blob/master/unstable/propertyvalue/propertyvalue.go#L21

Transform is also a walker if f does not edit the value. 

I know Go is a little bit more of a "let's do a little more code duplication than too much restrictive sharing" language so perhaps you looked there and it didn't fit the bill, but highlighting this.

Ideally the walkers would live in core alongside of PropertyValue definition itself.

**Review by t0yv0 — APPROVED** [review]

File: pkg/tfbridge/property_path.go (hunk 1 above)
Comment (VenelinMartinov) [reply pkg/tfbridge/property_path.go:54]: I didn't find anything like that so I wrote my own - happy to re-use any existing code here.

Comment (t0yv0) [reply pkg/tfbridge/property_path.go:54]: PTAL at the linked code and consider 🙏

Comment (VenelinMartinov) [reply pkg/tfbridge/property_path.go:54]: addressed in the previous PR in the stack: https://github.com/pulumi/pulumi-terraform-bridge/pull/2496

**Pull request merged**